tests/unit/lxd/test_launcher.py

```diff
@@ -143,10 +158,18 @@ def test_launch_no_base_instance(
     assert fake_instance.mock_calls == [
         call.exists(),
         call.launch(image="image-name", image_remote="image-remote", ephemeral=False),
+        call.config_get("user.craft_providers.status"),
+        call.config_get().__ne__("FINISHED"),
```

Comment:
This might be too brittle. It would probably be better to set `fake_instance.return_value` in the arrange space.

You can also test different parts of the loop by setting the fake instance's `side_effect` to a list of return values for successive calls.

Reply:
I am not sure why, but that is pytest output. And that are call checks not return values.

Reply:
`fake_instance` is a `Mock` object, so it's returning child `Mock` objects. If you set `fake_instance.config_get.return_value = "FINISHED"` before running the test, it shouldn't need this line.

Likewise, to check that the instance works correctly when the first query is in-progress but the second query is finished, you can set `fake_instance.config_get.side_effect = ["PREPARING", "FINISHED]`.

Reply:
Done